pkg/validations/upgradevalidations/immutableFields.go

```diff
@@ -88,26 +88,26 @@ func ValidateImmutableFields(ctx context.Context, k validations.KubectlClient, c
 		return fmt.Errorf("spec.externalEtcdConfiguration is immutable")
 	}
 
-	if !v1alpha1.RefSliceEqual(nSpec.IdentityProviderRefs, oSpec.IdentityProviderRefs) {
-		return fmt.Errorf("spec.identityProviderRefs is immutable")
+	oldAWSIamConfig := &v1alpha1.Ref{}
```

Comment:
can we rename this? `oldAWSIamConfigRef` same thing below. it's a bit unclear what object you define